### PR TITLE
some rotation fixes

### PR DESCRIPTION
**Pull request by jlipps**
@mykola-mokhnach these fixes resolved the issues i was having; what do you think?

### CODE REVIEW

File: WebDriverAgentLib/Commands/FBOrientationCommands.m [hunk 1]
```diff
@@ -50,7 +50,8 @@ + (NSArray *)routes
 + (id<FBResponsePayload>)handleGetOrientation:(FBRouteRequest *)request
 {
   FBSession *session = request.session;
-  return FBResponseWithStatus(FBCommandStatusNoError, [self.class interfaceOrientationForApplication:session.activeApplication]);
+  NSString *orientation = [self.class interfaceOrientationForApplication:session.activeApplication];
+  return FBResponseWithStatus(FBCommandStatusNoError, [[self _wdOrientationsMapping] valueForKey:orientation]);
```
Comment (jlipps) [inline WebDriverAgentLib/Commands/FBOrientationCommands.m]: note that this is a breaking change, but a breaking change to bring into line with the JSONWP. java client used to blanket return "unknown" for anything that wasn't PORTRAIT or LANDSCAPE anyway

File: WebDriverAgentLib/Commands/FBOrientationCommands.m [hunk 2]
```diff
@@ -65,7 +66,7 @@ + (NSArray *)routes
 + (id<FBResponsePayload>)handleGetRotation:(FBRouteRequest *)request
 {
     XCUIDevice *device = [XCUIDevice sharedDevice];
-    UIDeviceOrientation orientation = device.orientation;
+    UIInterfaceOrientation orientation = request.session.activeApplication.interfaceOrientation;
```
Comment (jlipps) [inline WebDriverAgentLib/Commands/FBOrientationCommands.m:69]: this is also a breaking change, but in my experience `device.orientation` always resulted in a 0-degree rotation, meaning it was useless.

File: WebDriverAgent.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -663,7 +663,7 @@
 		EE9AB75A1CAEDF0C008C271F /* FBInspectorCommands.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = FBInspectorCommands.h; sourceTree = "<group>"; };
 		EE9AB75B1CAEDF0C008C271F /* FBInspectorCommands.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = FBInspectorCommands.m; sourceTree = "<group>"; };
 		EE9AB75C1CAEDF0C008C271F /* FBOrientationCommands.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = FBOrientationCommands.h; sourceTree = "<group>"; };
-		EE9AB75D1CAEDF0C008C271F /* FBOrientationCommands.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = FBOrientationCommands.m; sourceTree = "<group>"; };
+		EE9AB75D1CAEDF0C008C271F /* FBOrientationCommands.m */ = {isa = PBXFileReference; fileEncoding = 4; indentWidth = 2; lastKnownFileType = sourcecode.c.objc; path = FBOrientationCommands.m; sourceTree = "<group>"; tabWidth = 2; };
```
Comment (mykola-mokhnach) [inline WebDriverAgent.xcodeproj/project.pbxproj]: Is this change needed? Can it be global?

File: WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme [hunk 4]
```diff
@@ -67,7 +67,7 @@
       <EnvironmentVariables>
          <EnvironmentVariable
             key = "USE_PORT"
-            value = "$(USE_PORT)"
+            value = "8100"
```
Comment (mykola-mokhnach) [inline WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme:70]: Is it just the default value or you just hardcoded it?

File: WebDriverAgentLib/Commands/FBOrientationCommands.m [hunk 5]
```diff
@@ -123,4 +124,20 @@ + (NSDictionary *)_orientationsMapping
   return orientationMap;
 }
 
++ (NSDictionary *)_wdOrientationsMapping
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Commands/FBOrientationCommands.m]: underscore is not needed - the method is anyway not visible from the outside of the implementation block

File: WebDriverAgentLib/Commands/FBOrientationCommands.m (hunk 1 above)
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Commands/FBOrientationCommands.m]: use `objectForKey`. `valueForKey` is used in KVO

File: WebDriverAgentLib/Commands/FBOrientationCommands.m [hunk 6]
```diff
@@ -123,4 +124,20 @@ + (NSDictionary *)_orientationsMapping
...
+{
+  static NSDictionary *orientationMap;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationMap =
+    @{
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Commands/FBOrientationCommands.m:139]: Can you please explain the purpose of such mapping?

File: WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme (hunk 4 above)
Comment (jlipps) [reply WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme:70]: i changed it from `$(USE_PORT)` to this. before, it was turning up as `0` anytime i tested the build.

probably there's some way to set the env var from outside xcode but i could not discover it. suggestions welcome.

File: WebDriverAgentLib/Commands/FBOrientationCommands.m (hunk 5 above)
Comment (jlipps) [reply WebDriverAgentLib/Commands/FBOrientationCommands.m]: i was matching style with `_orientationsMapping`, but i can not

File: WebDriverAgentLib/Commands/FBOrientationCommands.m (hunk 6 above)
Comment (jlipps) [reply WebDriverAgentLib/Commands/FBOrientationCommands.m:139]: added a comment in the code

**Comment by jlipps** [comment]
@mykola-mokhnach made some changes, take a look

**Review by mykola-mokhnach — APPROVED** [review]

File: WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme (hunk 4 above)
Comment (imurchie) [reply WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme:70]: This change breaks Appium's ability to change the WDA port. Another way to do it needs to be found.

Comment (jlipps) [reply WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme:70]: sorry, forgot this was still included in the merge. will revert.

Comment (jlipps) [reply WebDriverAgent.xcodeproj/xcshareddata/xcschemes/WebDriverAgentRunner.xcscheme:70]: https://github.com/appium/WebDriverAgent/pull/62
